framework/src/base/Moose.C

```diff
@@ -1112,10 +1106,27 @@ addActionTypes(Syntax & syntax)
                            "(check_integrity)");
 }
 
+void
+populateMeshOnlyTasks(Syntax & syntax)
+{
+  syntax.addDependencySets("(meta_action)"
```

Comment:
@YaqiWang - Here is my PR with the proposed MeshOnly changes. Here you will find the dependency sets in the Moose.C file were we can maintain them along with any other changes to the task lists.